### PR TITLE
Chat Widget Automation Send Message

### PR DESCRIPTION
**Pull request by surajkukkipady18**
### What do you want to achieve?
Automate Chat Widget Send Message Action
-

### PR Checklist

- [ ] I have tested it locally and all functionalities are working fine.
- [ ] I have compared it with mocks and all design elements are the same.
- [ ] I have tested it in IE Browser.

### How was the code tested?

-

### What new thing you came across while writing this code? 
-

### In case you fixed a bug then please describe the root cause of it? 
-

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: playwright_test/page/chatwidget.page.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import { URL_DETAILS } from "../utils/Url";
+import { APP_ID } from "../utils/Constant";
+
+class ChatWidgetPage {
+    constructor(page) {
+        this.page = page;
+
+        //Locators
+        this.testRadioBtn = page.locator("#test");
+        this.appIDbox = page.getByPlaceholder("23a6d30a59d247fa64605501f58074ea3");
```
Comment (devashishmamgain) [inline playwright_test/page/chatwidget.page.js:11]: replace it with constant

File: playwright_test/page/chatwidget.page.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        this.chatWidgetMsgSendBtn = this.kommunicateChatWidgetiframe.locator("#mck-msg-sbmt");
+        this.chatWidgetDeliveredIcon = this.kommunicateChatWidgetiframe.locator('xpath=//span[@title="delivered"]');
+    }
+
+    //Launch chat widget
+    async launchChatWidget() {
+        await this.page.goto(URL_DETAILS.CHATWIDGET_URL);
+        await this.testRadioBtn.click();
+        await this.appIDbox.click();
+        await this.appIDbox.press("Meta+a");
```
Comment (devashishmamgain) [inline playwright_test/page/chatwidget.page.js:25]: what is Meta+a ?

**Review by devashishmamgain — CHANGES_REQUESTED** [review]

File: playwright_test/page/chatwidget.page.js (hunk 2 above)
Comment (surajkukkipady18) [reply playwright_test/page/chatwidget.page.js:25]: Meta+a is used to do Select all (Control+A) action.

File: playwright_test/page/chatwidget.page.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+class ChatWidgetPage {
+    constructor(page) {
+        this.page = page;
+
+        //Locators
+        this.testRadioBtn = page.locator("#test");
+        this.appIDbox = page.getByPlaceholder(APP_ID.DEF_APP_ID);
+        this.launchWidgetBtn = page.getByRole("button", {name: "Launch Widget"});
+        this.kommunicateChatWidgetiframe = page.frameLocator('iframe[name="Kommunicate widget iframe"]');
```
Comment (Prochnost55) [inline playwright_test/page/chatwidget.page.js:13]: we can use id selector instead of name

File: playwright_test/page/chatwidget.page.js (hunk 1 above)
Comment (surajkukkipady18) [reply playwright_test/page/chatwidget.page.js]: Done.

File: playwright_test/tests-examples/demo-todo-app.spec.js [hunk 4]
```diff
@@ -0,0 +1,449 @@
+// @ts-check
+const { test, expect } = require('@playwright/test');
```
Comment (Prochnost55) [inline playwright_test/tests-examples/demo-todo-app.spec.js:2]: is this example from playwright?
is it required?

File: playwright_test/tests/chatwidgetlaunch.spec.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { test, expect } from "@playwright/test";
+import ChatWidgetPage from "../page/chatwidget.page";
+
+test("test", async ({ page }) => {
```
Comment (Prochnost55) [inline playwright_test/tests/chatwidgetlaunch.spec.js:4]: I guess here, "test" will be the name of test

shouldn't we use something more suitable for identification?

File: playwright_test/utils/Url.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export const URL_DETAILS = {
+    CHATWIDGET_URL: "https://test.prochnost.tk/widget-test",
```
Comment (Prochnost55) [inline playwright_test/utils/Url.js:2]: we can think of deploying this on one of our own domains

File: playwright_test/page/chatwidget.page.js (hunk 3 above)
Comment (surajkukkipady18) [reply playwright_test/page/chatwidget.page.js:14]: id is not present in project, have used xpath.

File: playwright_test/package.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  "name": "playwright_test",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {},
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@playwright/test": "^1.27.1"
```
[the remark this answers is not in the document]
Comment (surajkukkipady18) [reply playwright_test/package.json]: removed.

File: playwright_test/tests-examples/demo-todo-app.spec.js (hunk 4 above)
Comment (surajkukkipady18) [reply playwright_test/tests-examples/demo-todo-app.spec.js]: have deleted it.

File: playwright_test/tests/chatwidgetlaunch.spec.js (hunk 5 above)
Comment (surajkukkipady18) [reply playwright_test/tests/chatwidgetlaunch.spec.js]: Have renamed the test.

File: playwright_test/utils/Url.js (hunk 6 above)
Comment (surajkukkipady18) [reply playwright_test/utils/Url.js:2]: Will discuss with QA team and uodate.

**Review by devashishmamgain — APPROVED** [review]

**Review by Prochnost55 — APPROVED** [review]

**Review by vipinsonu1 — APPROVED** [review]

**Pull request merged**